mesa/agent.py

```diff
@@ -249,27 +249,38 @@ def do(
 
         return res if return_results else self
 
-    def get(self, attr_names: str | list[str]) -> list[Any]:
+    sentinel = object()
+
+    def get(self, attr_names: str | list[str], fallback_value=sentinel) -> list[Any]:
         """
         Retrieve the specified attribute(s) from each agent in the AgentSet.
 
         Args:
             attr_names (str | list[str]): The name(s) of the attribute(s) to retrieve from each agent.
+            fallback_value (Any, optional): The value to return if the attribute is not found. If not provided, an AttributeError is raised.
 
         Returns:
             list[Any]: A list with the attribute value for each agent in the set if attr_names is a str
             list[list[Any]]: A list with a list of attribute values for each agent in the set if attr_names is a list of str
 
         Raises:
-            AttributeError if an agent does not have the specified attribute(s)
-
+            AttributeError if an agent does not have the specified attribute(s) and no fallback_value is provided.
         """
 
+        def get_attr(agent, attr_name):
+            try:
+                return getattr(agent, attr_name)
+            except AttributeError:
```

Comment:
Catching an exception is comparatively costly in terms of performance. Normally this isn't something to worry about, but if we run into this frequently (which users might do) it could become a problem. I think it would be better to first check for the sentinel value and then either return `getattr(agent, attr_name)` or ` getattr(agent, attr_name, fallback_value)`.

Reply:
Hey facing some attribute error in the build after the changes, any ideas.....